sql/parser/eval_test.go

```diff
@@ -250,8 +250,8 @@ func TestEvalExpr(t *testing.T) {
 		{`'2010-09-28'::timestamp`, `2010-09-28 00:00:00+00:00`},
 		{`('2010-09-28 12:00:00.1'::timestamp)::date`, `2010-09-28`},
 		{`'2010-09-28 12:00:00.1'::timestamp`, `2010-09-28 12:00:00.1+00:00`},
-		{`'2010-09-28 12:00:00.1+02:00'::timestamp`, `2010-09-28 10:00:00.1+00:00`},
-		{`'2010-09-28 12:00:00.1-07:00'::timestamp`, `2010-09-28 19:00:00.1+00:00`},
+		{`'2010-09-28 12:00:00.1+02:00'::timestamp`, `2010-09-28 12:00:00.1+02:00`},
```

Comment:
On further reflection, I think that we should probably convert all input timestamps to the session time zone. We don't want to expose the timezone from which the timestamp was parsed, because that information will get lost if the value is written to a table and read back.